static/app/views/settings/account/accountSubscriptions.tsx

```diff
@@ -31,173 +36,162 @@ export type Subscription = {
   unsubscribedDate: string | null;
 };
 
-type State = DeprecatedAsyncView['state'] & {
-  subscriptions: Subscription[];
-};
+function AccountSubscriptions() {
+  const {data: subscriptions = [], isPending} = useApiQuery<Subscription[]>([ENDPOINT], {
+    staleTime: 2 * 60 * 1000,
+  });
+
+  const queryClient = useQueryClient();
+  const api = useApi();
+
+  const {mutate: updateSubscription} = useMutation({
+    mutationFn: (data: Subscription) =>
+      api.requestPromise(ENDPOINT, {
+        method: 'PUT',
+        data,
+      }),
+    onSuccess: (_resp, subscription: Subscription) => {
+      addSuccessMessage(
+        `${subscription.subscribed ? 'Subscribed' : 'Unsubscribed'} to ${subscription.listName}`
+      );
 
-class AccountSubscriptions extends DeprecatedAsyncView<
-  DeprecatedAsyncView['props'],
-  State
-> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    return [['subscriptions', ENDPOINT]];
+      // Update the subscription in the list
+      setApiQueryData<Subscription[]>(queryClient, [ENDPOINT], subs => {
+        return subs.map(sub => {
+          if (sub.listId === subscription.listId) {
+            return subscription;
+          }
+
+          return sub;
+        });
+      });
+    },
+    onError: (subscription: Subscription) => {
+      if (subscription) {
+        addErrorMessage(
+          `Unable to ${subscription.subscribed ? '' : 'un'}subscribe to ${subscription.listName}`
+        );
+      } else {
+        addErrorMessage('An unknown error occurred, please try again later.');
+      }
+    },
+  });
+
+  if (isPending) {
+    <LoadingIndicator />;
   }
 
-  getTitle() {
-    return t('Subscriptions');
-  }
+  const subGroups: Array<[string, Subscription[]]> = Object.entries(
+    subscriptions.reduce((acc, sub) => {
+      (acc[sub.email] = acc[sub.email] || []).push(sub);
+      return acc;
+    }, {})
```

Comment:
using `{}` as the type here only works because we have implicit any turned on
```suggestion
    subscriptions.reduce<Record<string, Subscription[]>((acc, sub) => {
      (acc[sub.email] = acc[sub.email] || []).push(sub);
      return acc;
    }, {})
```